proposals/0215-ios-replaykit-streaming.md

```diff
@@ -0,0 +1,102 @@
+                [self.rootViewController.view drawViewHierarchyInRect:bounds afterScreenUpdates:NO];
+            } break;
+        }
+        
+        UIImage *screenshot = UIGraphicsGetImageFromCurrentImageContext();
+        UIGraphicsEndImageContext();
+        
+        CGImageRef imageRef = screenshot.CGImage;
+        CVPixelBufferRef pixelBuffer = [self.class sdl_pixelBufferForImageRef:imageRef usingPool:self.streamManager.pixelBufferPool];
+        if (pixelBuffer != nil) {
+            [self.streamManager sendVideoData:pixelBuffer];
+            CVPixelBufferRelease(pixelBuffer);
+        }
+    };
+    
+}
+```
+
+## Potential downsides
+
+* ReplayKit displays an alert asking the user to allow the app to record the screen.  However, after the user taps this once, it will only ask again on a new app launch after a certain time window
+* The app would need to support horizontal orientation otherwise the screen would be captured in portrait
+* Must be on iOS 11 or greater
+
+## Impact on existing code
+
+* No changes to the Android library
+* The iOS library would need to alter it's code to add a flag to pick between ReplayKit streaming or the current method of streaming (possibly via the ```SDLStreamingMediaConfiguration``` ).  Additionally, it would need to alter SDLCarWindow.m's ```syncFrame``` method that causes the jump in CPU usage (which can be seen via the CPU profiling instrument while running an SDL enabled navigation app by filtering out system methods).  The app would need to be forced in a horizontal orientation.
+
+## Alternatives considered
+
+* There are private methods that can capture the screen very efficiently, however, implementing these can get your app removed from the app store or rejected from submission by Apple.
+
+* Nothing beyond keeping the original design.
+
+## Update
+
+* After testing the SDL Lock Screen appearance, ReplayKit does not take in a root veiw controller to display it seems to record the current window that is displayed.  Because the SDL Lockscreen shows up on another window, it does not seem to impede the recording of the original window.
+* Attached in proposal issue is an app that can be run as an example - there still needs to be code that resizes the pixel buffer (need to see if the is possible at all, the snippet provided did not alter the stream)
+* The app does not need to be landscaped if we can solve the pixel sizing issue, however, then we would also need to make sure the rootview controller does not resize on video streaming 
+* I determined the replay kit stream also contains audio data that can used to stream audio to the Head Unit exactly like a media app.  This would mean the developer would no longer need to send PCM audio files to SDL (they can play audio as they would in their app and it should carry over).
+* There would need tests to be done with iPhone XS, X, XS Max, XR - I am seeing a crash on these phones.
```

Comment:
Update and please refrain from using "I":
* Tests would need to be done with iPhone XS, X, XS Max, XR - when testing these phones, there has been notable crashes.